datafusion/common/src/table_reference.rs

```diff
@@ -450,3 +450,39 @@ mod tests {
         );
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum SchemaReference<'a> {
+    Bare {
+        schema: Cow<'a, str>,
+    },
+    Full {
+        schema: Cow<'a, str>,
+        catalog: Cow<'a, str>,
+    },
+}
+
+pub type OwnedSchemaReference = SchemaReference<'static>;
```

Comment:
Maybe we can also use this in `CreateCatalogSchema`?

Reply:
Good thinking!

Reply:
I actually think this might make more sense in a separate PR to reduce the surface area of code changed in this PR since this PR is already pretty big. I'm open to either though. What do you think? @viirya

Reply:
Sound good to me. It can be in a follow up.